core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -253,42 +260,58 @@ func (b *BulletproofTxManager) Trigger(addr common.Address) {
 	}
 }
 
+type NewTx struct {
+	FromAddress    common.Address
+	ToAddress      common.Address
+	EncodedPayload []byte
+	GasLimit       uint64
+	Meta           *EthTxMeta
+
+	MinConfirmations  null.Uint32
+	PipelineTaskRunID *uuid.UUID
+
+	Strategy TxStrategy
+}
+
 // CreateEthTransaction inserts a new transaction
-func (b *BulletproofTxManager) CreateEthTransaction(db *gorm.DB, fromAddress, toAddress common.Address, payload []byte, gasLimit uint64, meta interface{}, strategy TxStrategy) (etx EthTx, err error) {
-	err = CheckEthTxQueueCapacity(db, fromAddress, b.config.EvmMaxQueuedTransactions())
+func (b *BulletproofTxManager) CreateEthTransaction(db *gorm.DB, newTx NewTx) (etx EthTx, err error) {
+	err = CheckEthTxQueueCapacity(db, newTx.FromAddress, b.config.EvmMaxQueuedTransactions())
 	if err != nil {
 		return etx, errors.Wrap(err, "BulletproofTxManager#CreateEthTransaction")
 	}
 
-	// meta can hold arbitrary data and is mostly useful for logging/debugging
-	var metaBytes []byte
-	if meta != nil {
-		metaBytes, err = json.Marshal(meta)
-		if err != nil {
-			return etx, errors.Wrap(err, "BulletproofTxManager#CreateEthTransaction failed to marshal ethtx metadata")
-		}
-	}
-
 	value := 0
 	err = postgres.GormTransactionWithDefaultContext(db, func(tx *gorm.DB) error {
+		if newTx.PipelineTaskRunID != nil {
+			err = tx.Raw(`SELECT * FROM eth_txes WHERE pipeline_task_run_id = ?`, newTx.PipelineTaskRunID).Scan(&etx).Error
+			if err != nil {
+				return errors.Wrap(err, "BulletproofTxManager#CreateEthTransaction")
+			}
+
+			// if a previous transaction for this task run exists, immediately return it
+			if etx.ID != 0 {
+				return nil
+			}
+		}
+		// can't use DO NOTHING because it won't return any data, just use a no-op DO UPDATE instead
```

Comment:
perhaps comment needs to be updated? as it does not use `DO UPDATE`

Reply:
Good catch, I reverted the change but forgot to update the comment.